File: 69_Pizza/javascript/pizza.js
// PIZZA
//
// Converted from BASIC to Javascript by Oscar Toledo G. (nanochess)
//

function print(str)
{
    document.getElementById("output").appendChild(document.createTextNode(str));
}

function input()
{
    var input_element;
    var input_str;
    
    return new Promise(function (resolve) {
                       input_element = document.createElement("INPUT");
                       
                       print("? ");
                       input_element.setAttribute("type", "text");
                       input_element.setAttribute("length", "50");
                       document.getElementById("output").appendChild(input_element);
                       input_element.focus();
                       input_str = undefined;
                       input_element.addEventListener("keydown", function (event) {
                                                      if (event.keyCode == 13) {
                                                      input_str = input_element.value;
                                                      document.getElementById("output").removeChild(input_element);
                                                      print(input_str);
                                                      print("\n");
                                                      resolve(input_str);
                                                      }
                                                      });
                       });
}

function tab(space)
{
    var str = "";
    while (space-- > 0)
        str += " ";
    return str;
}

var sa = [, "A","B","C","D","E","F","G","H","I","J","K","L","M","N","O","P"];
var ma = [, "1","2","3","4"];
var a = [];

// Main program
async function main()
{
    print(tab(33) + "PIZZA\n");
    print(tab(15) + "CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n");
    print("\n");
    print("\n");
    print("\n");
    print("PIZZA DELIVERY GAME\n");
    print("\n");
    print("WHAT IS YOUR FIRST NAME");
    ns = await input();
    print("\n");
    print("HI, " + ns + ". IN THIS GAME YOU ARE TO TAKE ORDERS\n");
    print("FOR PIZZAS.  THEN YOU ARE TO TELL A DELIVERY BOY\n");
    print("WHERE TO DELIVER THE ORDERED PIZZAS.\n");
    print("\n");
    print("\n");
    print("MAP OF THE CITY OF HYATTSVILLE\n");
    print("\n");
    print(" -----1-----2-----3-----4-----\n");
    k = 4;
    for (i = 1; i <= 4; i++) {
        print("-\n");
        print("-\n");
        print("-\n");
        print("-\n");
        print(ma[k]);
        s1 = 16 - 4 * i + 1;
        print("     " + sa[s1] + "     " + sa[s1 + 1] + "     " + sa[s1 + 2] + "     ");
        print(sa[s1 + 3] + "     " + ma[k] + "\n");
        k--;
    }
    print("-\n");
    print("-\n");
    print("-\n");
    print("-\n");
    print(" -----1-----2-----3-----4-----\n");
    print("\n");
    print("THE OUTPUT IS A MAP OF THE HOMES WHERE\n");
    print("YOU ARE TO SEND PIZZAS.\n");
    print("\n");
    print("YOUR JOB IS TO GIVE A TRUCK DRIVER\n");
    print("THE LOCATION OR COORDINATES OF THE\n");
    print("HOME ORDERING THE PIZZA.\n");
    print("\n");
    while (1) {
        print("DO YOU NEED MORE DIRECTIONS");
        str = await input();
        if (str == "YES" || str == "NO")
            break;
        print("'YES' OR 'NO' PLEASE, NOW THEN, ");
    }
    if (str == "YES") {
        print("\n");
        print("SOMEBODY WILL ASK FOR A PIZZA TO BE\n");
        print("DELIVERED.  THEN A DELIVERY BOY WILL\n");
        print("ASK YOU FOR THE LOCATION.\n");
        print("     EXAMPLE:\n");
        print("THIS IS J.  PLEASE SEND A PIZZA.\n");
        print("DRIVER TO " + ns + ".  WHERE DOES J LIVE?\n");
        print("YOUR ANSWER WOULD BE 2,3\n");
        print("\n");
        print("UNDERSTAND");
        str = await input();
        if (str != "YES") {
            print("THIS JOB IS DEFINITELY TOO DIFFICULT FOR YOU. THANKS ANYWAY");
            return;
        }
        print("GOOD.  YOU ARE NOW READY TO START TAKING ORDERS.\n");
        print("\n");
        print("GOOD LUCK!!\n");
        print("\n");
    }
    while (1) {
        for (i = 1; i <= 5; i++) {
            s = Math.floor(Math.random() * 16 + 1);
            print("\n");
            print("HELLO " + ns + "'S PIZZA.  THIS IS " + sa[s] + ".\n");
            print("  PLEASE SEND A PIZZA.\n");
            while (1) {
                print("  DRIVER TO " + ns + ":  WHERE DOES " + sa[s] + " LIVE");
                str = await input();
                a[1] = parseInt(str);
                a[2] = parseInt(str.substr(str.indexOf(",") + 1));
                t = a[1] + (a[2] - 1) * 4;
                if (t != s) {
                    print("THIS IS " + sa[t] + ". I DID NOT ORDER A PIZZA.\n");
                    print("I LIVE AT " + a[1] + "," + a[2] + "\n");
                } else {
                    break;
                }
            }
            print("HELLO " + ns + ".  THIS IS " + sa[s] + ", THANKS FOR THE PIZZA.\n");
        }
        print("\n");
        print("DO YOU WANT TO DELIVER MORE PIZZAS");
        str = await input();
        if (str != "YES")
            break;
    }
    print("\n");
    print("O.K. " + ns + ", SEE YOU LATER!\n");
    print("\n");
}

main();

File: 69_Pizza/perl/pizza.pl
#!/usr/bin/perl
#use strict;
# Automatic converted by bas2perl.pl

print ' 'x33 . "PIZZA". "\n";
print ' 'x15 . "CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY". "\n";
print "\n"; print "\n"; print "\n";
my @S; my @M;
print "PIZZA DELIVERY GAME". "\n"; print "\n";
print "WHAT IS YOUR FIRST NAME? "; chomp($N = uc(<STDIN>)); print "\n";
print "HI, ". $N. ".  IN THIS GAME YOU ARE TO TAKE ORDERS". "\n";
print "FOR PIZZAS.  THEN YOU ARE TO TELL A DELIVERY BOY". "\n";
print "WHERE TO DELIVER THE ORDERED PIZZAS.". "\n"; print "\n"; print "\n";
for ($I=1; $I<=16; $I++) {
$S[$I]= <DATA>; chomp($S[$I]);;
}
for ($I=1; $I<=4; $I++) {
$M[$I]= <DATA>; chomp($M[$I]);;
}
# TO DATA SEGMENT;
# TO DATA SEGMENT;
print "MAP OF THE CITY OF HYATTSVILLE". "\n"; print "\n";
print " -----1-----2-----3-----4-----". "\n";
$K=4;
for ($I=1; $I<=4; $I++) {
print "-". "\n"; print "-". "\n"; print "-". "\n"; print "-". "\n";
print $M[$K];
$S1=16-4*$I+1;
print "     ". $S[$S1]. "     ". $S[$S1+1]. "     ". $S[$S1+2]. "     ";
print $S[$S1+3]. "     ". $M[$K]. "\n";
$K=$K-1;
}
print "-". "\n"; print "-". "\n"; print "-". "\n"; print "-". "\n";
print " -----1-----2-----3-----4-----". "\n"; print "\n";
print "THE OUTPUT IS A MAP OF THE HOMES WHERE". "\n";
print "YOU ARE TO SEND PIZZAS.". "\n"; print "\n";
print "YOUR JOB IS TO GIVE A TRUCK DRIVER". "\n";
print "THE LOCATION OR COORDINATES OF THE". "\n";
print "HOME ORDERING THE PIZZA.". "\n"; print "\n";

Line520:
print "DO YOU NEED MORE DIRECTIONS? "; chomp($A = uc(<STDIN>));
if ($A eq "YES") { goto Line590; }
if ($A eq "NO") { goto Line750; }
print "'YES' OR 'NO' PLEASE, NOW THEN,". "\n"; goto Line520;

Line590:
print "\n"; print "SOMEBODY WILL ASK FOR A PIZZA TO BE". "\n";
print "DELIVERED.  THEN A DELIVERY BOY WILL". "\n";
print "ASK YOU FOR THE LOCATION.". "\n"; print "     EXAMPLE:". "\n";
print "THIS IS J.  PLEASE SEND A PIZZA.". "\n";
print "DRIVER TO ". $N. ".  WHERE DOES J LIVE?". "\n";
print "YOUR ANSWER WOULD BE 2,3". "\n"; print "\n";
print "UNDERSTAND? "; chomp($A = uc(<STDIN>));
if ($A eq "YES") { goto Line690; }
print "THIS JOB IS DEFINITELY TOO DIFFICULT FOR YOU. THANKS ANYWAY". "\n";
goto Line999;

Line690:
print "GOOD.  YOU ARE NOW READY TO START TAKING ORDERS.". "\n"; print "\n";
print "GOOD LUCK!!". "\n"; print "\n";

Line750:
for ($I=1; $I<=5; $I++) {
$S=int(rand(1)*16+1); print "\n";
print "HELLO ". $N. "'S PIZZA.  THIS IS ". $S[$S]. ".";
print "  PLEASE SEND A PIZZA.". "\n";

Line780:
print "  DRIVER TO ". $N. ":  WHERE DOES ". $S[$S]. " LIVE";
print "? "; chomp($Inp_ = uc(<STDIN>)); ($A[1],$A[2])= split(/,/, $Inp_);
$T=$A[1]+($A[2]-1)*4;
if ($T eq $S) { goto Line920; }
print "THIS IS ". $S[$T]. ".  I DID NOT ORDER A PIZZA.". "\n";
print "I LIVE AT ". $A[1]. ",". $A[2]. "\n";
goto Line780;

Line920:
print "HELLO ". $N. ".  THIS IS ". $S[$S]. ", THANKS FOR THE PIZZA.". "\n";
}
print "\n"; print "DO YOU WANT TO DELIVER MORE PIZZAS? "; chomp($A = uc(<STDIN>));
if ($A eq "YES") { goto Line750; }
print "\n"; print "O.K. ". $N. ", SEE YOU LATER!". "\n"; print "\n";

Line999:
exit;



__DATA__
A
B
C
D
E
F
G
H
I
J
K
L
M
N
O
P
1
2
3
4
